File: app/web_searcher.py
import requests
from bs4 import BeautifulSoup

def search_web(query):
    """Perform a simple web search and extract a clean summary."""
    headers = {
        "User-Agent": "Mozilla/5.0"
    }
    try:
        response = requests.get(f"https://www.google.com/search?q={query}", headers=headers)
        soup = BeautifulSoup(response.text, "html.parser")

        # Try to find the featured snippet or top result
        answer = soup.find("div", class_="BNeawe").text
        return answer.strip() if answer else "Sorry, I couldn't find anything useful."
    except Exception as e:
        return f"Error during web search: {str(e)}"

File: src/app/WebSearcher.java
package app;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.select.Elements;

public class WebSearcher {
    public static String search(String query) {
        try {
            String url = "https://en.wikipedia.org/wiki/" + query.replace(" ", "_");
            Document doc = Jsoup.connect(url).get();
            Elements paragraphs = doc.select("p");
            return paragraphs.first().text();
        } catch (Exception e) {
            return "Sorry, I couldn't find information.";
        }
    }
}
